recipes/armadillo/all/conanfile.py

```diff
@@ -0,0 +1,489 @@
+            and not self.options.USE_SYSTEM_BLAS
+        ):
+            blas_conflicts = [
+                x for x in self.exclusions["blas"] if getattr(self.options, x)
+            ]
+            if (
+                (
+                    (
+                        self.settings.os == "Macos"
+                        and self.options.ALLOW_BLAS_LAPACK_MACOS
+                        and self.options.ALLOW_OPENBLAS_MACOS
+                    )
+                    or self.settings.os != "Macos"
+                )
+                and self.options.ARMA_USE_BLAS
+                and not blas_conflicts
+            ):
+                self.options.USE_OPENBLAS = True
+        if self.options.ALLOW_BLAS_LAPACK_MACOS:
+            self.options.ARMA_USE_ACCELERATE = False
+
+    def validate(self):
+        tools.check_min_cppstd(self, "11")
+        # Iterate over dependency maps to confirm requirements are met
+        for option, dependency in self.os_dependencies.items():
+            if getattr(self.options, option) and self.settings.os != dependency:
+                raise ConanInvalidConfiguration(
+                    f"Option {option} can only be enabled on {dependency} operating systems"
+                )
+        # Optional option co-depdendencies
+        for platform, opts in self.opt_dependencies.items():
+            if platform == "all" or platform == self.settings.os:
+                for option, dependencies in opts.items():
+                    dependencies_in_error = [
+                        dependency
+                        for dependency in dependencies
+                        if not getattr(self.options, dependency)
+                    ]
+                    if getattr(self.options, option) and dependencies_in_error:
+                        raise ConanInvalidConfiguration(
+                            "Option {option} cannot be enabled without enabling {missing}".format(
+                                option=option, missing=", ".join(dependencies_in_error)
+                            )
+                        )
+        # Mutually exclusive options
+        for category, incompatible_opts in self.exclusions.items():
+            conflicts = [x for x in incompatible_opts if getattr(self.options, x)]
+            if len(conflicts) > 1:
+                raise ConanInvalidConfiguration(
+                    "The following options are in conflict: {conflict}. Resolve this conflict by ensuring only one of these options is enabled and try again.".format(
+                        conflict=", ".join(conflicts)
+                    )
+                )
+        if self.settings.compiler == "Visual Studio" and self.options.shared:
+            self.output.warn(
+                "Building a shared library with MSVC is not supported. This may not be successful."
+            )
+
+    def requirements(self):
+        # Optional requirements
+        openblas = "openblas/[>=0.3.10]"
+        hdf5 = "hdf5/[>=1.12.0]"
```

Comment:
I'm not sure version ranges are allowed on CCI

Reply:
No, do not use version ranges in CCI, please.

If using version ranges, a new package ID might be computed for a new version of the requirements and all the graph of dependencies will start to fail with a _missing packages_ error.